java/src/test/java/ai/rapids/cudf/TableTest.java

```diff
@@ -3182,88 +3302,185 @@ void testWindowingLag() {
+                 new StructData(-11, "s11"), null, new StructData(-13, "s13"), new StructData(-14, "s14"),
+                 new StructData(-111, "s111"), new StructData(null, "s112"), new StructData(-222, "s222"), new StructData(-333, "s333"));
+             Table windowAggResults = sorted.groupBy(0, 1)
+                 .aggregateWindows(Aggregation
+                     .lag(1, defaultOutput)
+                     .onColumn(3)
+                     .overWindow(options));
+             Table decWindowAggResults = sorted.groupBy(0, 4)
+                 .aggregateWindows(Aggregation
+                     .lag(1, decDefaultOutput)
+                     .onColumn(6)
+                     .overWindow(options));
+             Table listWindowAggResults = sorted.groupBy(0, 1).aggregateWindows(
+                 Aggregation
+                     .lag(1, listDefaultOutput)
+                     .onColumn(7)
+                     .overWindow(options));
+             Table structWindowAggResults = sorted.groupBy(0, 1).aggregateWindows(
+                 Aggregation
+                     .lag(1, structDefaultOutput)
+                     .onColumn(8)
+                     .overWindow(options));
+             ColumnVector expectAggResult = ColumnVector.fromBoxedInts(0, 7, 5, 1, -4, 7, 9, 8, -8, 8, 0, 6);
+             ColumnVector decExpectAggResult = ColumnVector.decimalFromLongs(-2, 0, 7, 5, 1, -4, 7, 9, 8, -8, 8, 0, 6);
+             ColumnVector listExpectAggResult = ColumnVector.fromLists(
+                 new HostColumnVector.ListType(true, new HostColumnVector.BasicType(true, DType.INT32)),
+                 Arrays.asList(111), Arrays.asList(11, 12, null, 13), Arrays.asList(14, null, 15, null), Arrays.asList((Integer) null),
+                 Arrays.asList(-11), Arrays.asList(21, null, null, 22), Arrays.asList(23, 24), Arrays.asList(25, 26, 27),
+                 Arrays.asList(6), Arrays.asList(null, 31), Arrays.asList(32, 33, 34), Arrays.asList(35, 36));
+             ColumnVector structExpectAggResult = ColumnVector.fromStructs(
+                 new StructType(true,
+                     new BasicType(true, DType.INT32),
+                     new BasicType(true, DType.STRING)),
+                 new StructData(-1, "s1"), new StructData(1, "s1"), new StructData(null, "s2"), new StructData(2, null),
+                 new StructData(-11, "s11"), new StructData(11, "s11"), null, new StructData(13, "s13"),
+                 new StructData(-111, "s111"), new StructData(111, "s111"), new StructData(null, "s112"), new StructData(2, "s222"))) {
+          assertColumnsAreEqual(expectAggResult, windowAggResults.getColumn(0));
+          assertColumnsAreEqual(decExpectAggResult, decWindowAggResults.getColumn(0));
+          assertColumnsAreEqual(listExpectAggResult, listWindowAggResults.getColumn(0));
+          // TODO  this is not gonna work, since cudf has some issue on lag with default values
```

Comment:
Please indicate that the problem is for `STRUCT` default columns. It looks like things work for lists.
We can also mention the issue here, once it is raised.

Reply:
Done